eldritch/stories.py

```diff
@@ -0,0 +1,402 @@
+  def get_max_token_event(self, token_type, owner):
+    if token_type == "clue":
+      return self.advance_story(owner, True)
+    return events.Nothing()
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.Bless):
+      return events.AddToken(self, "clue", owner)
+    if (
+        isinstance(event, events.BlessCurse)
+        and (event.character == owner)
+        and event.is_resolved()
+        and values.ItemNameCount(owner, "Curse").value(state) == 1
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class PhotographerStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("There's Your Proof")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.DrawSpecific(owner, "specials", "Retainer")
+
+  def get_interrupt(self, event, owner, state):
+    if (
+        isinstance(event, events.AddToken)
+        and event.character == owner
+        and event.token_type == "must_roll"
+        and event.asset.name == "Retainer"
+    ):
+      return events.CancelEvent(event)
+    return None
+
+
+class PhotographerStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("The Film Is Ruined")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Sequence([
+        events.DiscardNamed(owner, "Retainer"),
+        events.DrawSpecific(owner, "specials", "Bad Credit")
+    ], owner
+    )
+
+  def get_override(self, other, attribute):
+    if isinstance(other, (assets.BankLoan, assets.Retainer)) and attribute == "can_keep":
+      return False
+    return None
+
+
+class PhotographerStory(Story):
+  def __init__(self):
+    super().__init__("A Thousand Words", "There's Your Proof", "The Film Is Ruined")
+
+  def get_trigger(self, event, owner, state):
+    if len([t for t in owner.trophies if isinstance(t, gates.Gate)]) >= 2:
+      return self.advance_story(owner, True)
+    if owner.clues >= 5:
+      return self.advance_story(owner, False)
+    return None
+
+
+class PsychologistStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("Cured")
+
+  def get_interrupt(self, event, owner, state):
+    return events.Sequence([
+        events.Bless(owner),
+        events.Gain(owner, {"clues": 3})
+    ])
+
+
+class PsychologistStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("Followed", {"sneak": -1})
+
+
+class PsychologistStory(Story):
+  def __init__(self):
+    super().__init__("Final Analysis", "Cured", "Followed")
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.GainOrLoss) and owner.clues >= 6:
+      return self.advance_story(owner, False)
+    return None
+
+  def get_usable_interrupt(self, event, owner: characters.Character, state):
+    if (
+        isinstance(event, events.Upkeep)
+        and event.character == owner
+        and owner.place.name == "Asylum"
+    ):
+      choice = events.SpendChoice(
```

Comment:
Might make this a CardSpendChoice (where one of the choices is the name of the card) so that the story shows up in the user interface.